MatrixSDK/Data/RoomSummaryStore/CoreData/Models/MXRoomLastMessageMO.swift

```diff
@@ -47,19 +47,15 @@ public class MXRoomLastMessageMO: NSManagedObject {
         s_originServerTs = lastMessage.originServerTs
         s_isEncrypted = lastMessage.isEncrypted
         s_sender = lastMessage.sender
-        s_text = lastMessage.text
         
-        if let attributedText = lastMessage.attributedText {
-            s_attributedText = NSKeyedArchiver.archivedData(withRootObject: attributedText)
-        } else {
-            s_attributedText = nil
-        }
+        s_text = lastMessage.isEncrypted ? nil : lastMessage.text
         
-        if let others = lastMessage.others {
-            s_others = NSKeyedArchiver.archivedData(withRootObject: others)
+        if lastMessage.isEncrypted {
+            s_attributedText = lastMessage.encryptedAttributedString()
         } else {
-            s_others = nil
+            s_attributedText = lastMessage.attributedText.map(NSKeyedArchiver.archivedData(withRootObject:))
```

Comment:
I think its a bit confusing that archiving of encrypted data happens in `MXRoomLastMessage` but unencrypted in `MXRoomLastMessageMO` ... it creates some assumptions about responsibilities that are not obvious from the API.

An alternative approach would be to simply expose `archivedAttributedString` getter on `MXRoomLastMessage` which would be used by `MXRoomLastMessageMO` directly. That way the latter will not need to perform any additional encryption checks